cassovary-core/src/main/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraph.scala

```diff
@@ -153,18 +168,24 @@ class BipartiteGraph(val leftNodes: Array[BipartiteNode], val leftNodeCount: Int
 
     if (id == 0) {
       return None // id 0 is not allowed in bipartite graph
-    } else if (isLeftNodeId) {
-      //invert nodeId to index
-      val leftOriginalId = - id
-      if (leftOriginalId < leftNodes.length) {
-        if (leftNodes(leftOriginalId) != null) Some(leftNodes(leftOriginalId))
-        else None
-      } else None
+    } else if (!nodeIdsUnique) {
```

Comment:
this should be before any other code here, as id==0 is also possible in the other case